chrome/content/zotero/xpcom/attachments.js

```diff
@@ -1105,9 +1106,91 @@ Zotero.Attachments = new function(){
 			});
 			
 			if (options.isPDF) {
+				enforcingPDF = true;
 				await _enforcePDF(path);
 			}
 		}
+		catch (e) {
+			try {
+				await OS.File.remove(path, { ignoreAbsent: true });
+			}
+			catch (e) {
+				Zotero.debug(e, 1);
+			}
+			// Custom handling for PDFs that are bot-guarded
+			// via a JS-redirect
+			if (enforcingPDF && e instanceof this.InvalidPDFException
+					&& url.startsWith('https://www.sciencedirect.com/')) {
+				this.downloadPDFViaBrowser(url, path, options);
+			}
+			else {
+				throw e;
+			}
+		}
+	};
+
+	/**
+	 * @param {String} url
+	 * @param {String} path
+	 * @param {Object} [options]
+	 * @param {Object} [options.cookieSandbox]
+	 */
+	this.downloadPDFViaBrowser = async function (url, path, options = {}) {
+		Zotero.debug(`downloadPDFViaBrowser: Downloading file via browser from ${url}`);
+		const timeout = 60e3;
+		let channelBrowser, hiddenBrowser;
+		let hiddenBrowserPDFFoundDeferred = Zotero.Promise.defer();
+		
+		var pdfMIMETypeHandler = async (data, name, _, channel) => {
+			Zotero.debug(`downloadPDFViaBrowser: Sniffing a PDF loaded at ${name}`);
+			Zotero.debug(data.substr(0, 10));
+			
+			let isOurPDF = false;
+			// try the browser
+			try {
+				channelBrowser = channel.notificationCallbacks.getInterface(Ci.nsIWebNavigation)
+					.QueryInterface(Ci.nsIDocShell).chromeEventHandler;
+			}
+			catch (e) {}
+			if (channelBrowser) {
+				isOurPDF = hiddenBrowser === channelBrowser;
+			}
+			else {
+				// try the document for the load group
```

Comment:
Can you clarify when this would be relevant?

Reply:
I have no clue. This is taken from, but there must be a reason. Git blame does not reveal it though. https://github.com/zotero/zotero/blob/61e976bd3aec95b73cd993e685adfa897ab3784f/chrome/content/zotero/xpcom/cookieSandbox.js#L331-L360